content/articles/build-a-portfolio-website-with-vite-and-threejs/index.md

```diff
@@ -0,0 +1,449 @@
+Let’s check our browser
+
+![torus](/engineering-education/build-a-portfolio-website-with-vite-and-threejs/torus-shape.png)
+
+Our torus has no movement, so let's change that. Go back to our animation loop and add this:
+
+```javascript
+ torus.rotation.x += 0.01;
+ torus.rotation.y += 0.005;
+ torus.rotation.y += 0.01;
+```
+
+Every shape used has a different position, rotation, and scale properties. What we did here was change the rotation along the x-axis by 0.01. We also did the same for the y and z-axis (changing the rotation along their axis). Let’s go back to our app:
+
+![torus_round](/engineering-education/build-a-portfolio-website-with-vite-and-threejs/torus-round.gif)
+
+As you can see, it’s animating in an infinite loop. Pretty cool, right? Now we want to introduce the concept of lightning. Lightening is what makes your 3d objects come to life. Let’s go back to our material. Change from `MeshBasicMaterial` to `MeshStandardMaterial`. This is a material that will react to light bouncing off it. Let’s also remove the `wireframe` property, so our material will look like this:
+
+```javascript
+const material =  new THREE.MeshStandardMaterial({color:0xFF6347});
+```
+
+Let’s go back to our browser. You’ll notice we have a black screen again. This is because we don't have anything lightening our object. So let’s create one. Paste this code right after the `scene.add(torus)`:
+
+```javascript
+scene.add(torus)
+const pointLight = new THREE.PointLight(0xffffff)
+pointLight.position.set(5,5,5)
+ const ambientLight = new THREE.AmbientLight(0xffffff);
+ scene.add(pointLight, ambientLight)
+ ```
+
+Here we are instantiating a new `pointLight` with a color of white, setting its values, and adding it to our scene. We also instantiated an ambient light. It will give lightening to the entire scene equally.
+
+Let’s add a `lightHelper` and a `gridHelper`. A `lightHelper` is used to show the position of a `pointLight`. While the `gridHelper` draws a two-dimensional grid. So let’s add it to the scene:
+
+```javascript
+const lightHelper = new THREE.PointLightHelper(pointLight)
+ const gridHelper = new THREE.GridHelper(200,50);
+ scene.add(lightHelper)
+ ```
+ 
+Let’s add orbit controls to it. This will allow us to move around the scene using our mouse. To do that, we first have to import the orbits control from Three.js:
+
+```javascript
+import {OrbitControls} from 'three/examples/jsm/controls/OrbitControls';
+```
+
+Once we’ve done that we now have to instantiate the controls class and pass in our arguments. Right after our `scene.add` we input the following:
+```javascript
+const controls = new OrbitControls(camera,renderer.domElement);
+```
+
+This would listen to `dom` events on the mouse and update the camera position accordingly.
+We then need to call a `controls.update` in the animation loop to ensure that the changes are reflected in the UI.Inside our `requestAnimationFrame` add this:
+
+```javascript
+controls.update();
+```  
```

Comment:
Is this line of code supposed to be inside our `requestAnimationFrame` or the function `animate()`?